scene/2d/physics/collision_polygon_2d.cpp

```diff
@@ -292,6 +292,17 @@ real_t CollisionPolygon2D::get_one_way_collision_margin() const {
 	return one_way_collision_margin;
 }
 
+void CollisionPolygon2D::set_one_way_collision_direction(const Vector2 &p_direction) {
```

Comment:
Actually I just realized that you are not updating one-way preview for polygons. You need to add `queue_redraw()` here and the same check as in CollisionShape2D. Also the arrow needs to be rotated.
![image](https://github.com/user-attachments/assets/450385e9-c563-4e20-bb25-0a6c6b3ee332)

Reply:
Good catch, I'll give this a try.

Reply:
You still need to add a check whether new direction is different.